src/__swaps__/screens/Swap/hooks/useEstimatedGasFee.ts

```diff
@@ -35,20 +43,68 @@ export function useEstimatedGasFee({
   return useMemo(() => {
     if (!gasLimit || !gasSettings || !nativeNetworkAsset?.price) return;
 
-    const amount = gasSettings.isEIP1559 ? add(gasSettings.maxBaseFee, gasSettings.maxPriorityFee) : gasSettings.gasPrice;
-
-    const totalWei = multiply(gasLimit, amount);
+    const fee = calculateGasFee(gasSettings, gasLimit);
     const networkAssetPrice = nativeNetworkAsset.price.value?.toString();
 
-    if (!networkAssetPrice) return `${formatNumber(weiToGwei(totalWei))} Gwei`;
+    if (!networkAssetPrice) return `${formatNumber(weiToGwei(fee))} Gwei`;
 
-    const gasAmount = formatUnits(safeBigInt(totalWei), nativeNetworkAsset.decimals).toString();
-    const feeInUserCurrency = multiply(networkAssetPrice, gasAmount);
+    const feeFormatted = formatUnits(safeBigInt(fee), nativeNetworkAsset.decimals).toString();
+    const feeInUserCurrency = multiply(networkAssetPrice, feeFormatted);
 
     return formatCurrency(feeInUserCurrency);
   }, [gasLimit, gasSettings, nativeNetworkAsset]);
 }
 
+const getUserHasEnoughFundsForTx = (quote: Quote, gasFee: string) => {
+  const nativeAsset = getUserNativeNetworkAsset(quote.chainId);
+  if (!nativeAsset) return false;
+  const userBalance = nativeAsset.balance.amount || '0';
+
+  const nativeAmountSelling = isSameAddress(nativeAsset.address, quote.sellTokenAddress) ? quote.sellAmount.toString() : '0';
+  const totalNativeSpentInTx = formatUnits(
+    safeBigInt(add(add(quote.value?.toString() || '0', gasFee), nativeAmountSelling)),
+    nativeAsset.decimals
+  );
+
+  return lessThan(totalNativeSpentInTx, userBalance);
+};
+
+function useSyncSwapTransactionSharedValues({
+  quote,
+  gasSettings,
+  estimatedGasLimit,
+}: {
+  quote: Quote | CrosschainQuote | QuoteError | null;
+  gasSettings: GasSettings | undefined;
+  estimatedGasLimit: string | undefined;
+}) {
+  const {
+    estimatedGasLimit: estimatedGasLimitSharedValue,
+    gasSettings: gasSettingsSharedValue,
+    userHasEnoughFundsForTx,
+  } = useSwapContext();
+
+  useEffect(() => {
+    if (!gasSettings || !estimatedGasLimit || !quote || 'error' in quote) {
+      estimatedGasLimitSharedValue.value = undefined;
+      gasSettingsSharedValue.value = undefined;
+      return;
+    }
+
+    estimatedGasLimitSharedValue.value = estimatedGasLimit;
+    gasSettingsSharedValue.value = gasSettings;
+
+    const gasFee = calculateGasFee(gasSettings, estimatedGasLimit);
+
+    userHasEnoughFundsForTx.value = getUserHasEnoughFundsForTx(quote, gasFee);
```

Comment:
note for @benisgold we should make sure this is in line with the changes he's making for the additional buffer of gas when calculating max + gas for native tokens